### PR TITLE
Hui bugfix 65 "Students unable to submit assignment after a recent change"

### PR DESCRIPTION
**Pull request by spm2020spring**
The following bug is temporarily fixed.  Now students should be able to submit Quiz 5 (Individual assignment).

http://118.25.96.118/bugzilla/show_bug.cgi?id=65

This fix does not work for Group assignment.  Therefore, $group_id must be properly set in the future.

**Hui**

### CODE REVIEW

File: Script.php [hunk 1]
```diff
@@ -827,7 +827,13 @@ function checksize($file)
      if ($con->query($sql1) === TRUE) {
      }
      
-    
+
+    // When $group_id is not properly initialized, use integer 0 as its value.
+    // This temporarily fixed the "Students unable to submit assignment after a recent change" bug at http://118.25.96.118/bugzilla/show_bug.cgi?id=65
+    if (trim($group_id) === '') {
```
Comment (Teecloudy) [inline Script.php]: @spm2020spring 
This worked, the Quiz 5 was submitted successfully. What does trim() do?

**Ashly**

**Review by Teecloudy — COMMENTED** [review]
@spm2020spring 

Great Fix (Temporarily).
Why is the solution temporary and why doesn't it support Groups submissions.

**Ashly**

**Review by Teecloudy — APPROVED** [review]

File: Script.php (hunk 1 above)
Comment (spm2020spring) [reply Script.php]: > @spm2020spring
> This worked, the Quiz 5 was submitted successfully. What does trim() do?
> 
> **Ashly**

@Teecloudy 

The function trim() removes all whitespaces.  

Therefore,  trim($group_id) will always return an empty string `''` if $group_id does not contain non-whitespace characters.  

**Hui**

**Comment by spm2020spring** [comment]
> @spm2020spring
> 
> Great Fix (Temporarily).
> Why is the solution temporary and why doesn't it support Groups submissions.
> 
> **Ashly**

@Teecloudy 

If this assignment is of type **Group**,  each group probably has a non-zero group ID.

Therefore, it is not correct to always set $group_id 0.   That is why I emphasized that
$group_id must be set properly before the insertion, depending on the assignment type.

**Hui**

**Pull request merged**